### PR TITLE
[NNPA] Add ExpandAttentionMask Pass for NNPA Broadcasting Support

### PR DESCRIPTION
**Pull request by tungld**
This PR implements a new ONNX transformation pass to address NNPA broadcasting limitations in attention mechanisms by expanding attention masks before MatMul-Add-Softmax patterns.

See #3640.

**Core Implementation:**
- `src/Dialect/ONNX/Transforms/ExpandAttentionMask.cpp`
  - New pass that detects MatMul_Add_Softmax patterns
  - Expands broadcasted attention masks to full shape using onnx.Expand
  - Integrates with DimAnalysis for dynamic shape handling
  - Processes multiple patterns in a single function

**Pass Infrastructure:**
- `src/Dialect/ONNX/Transforms/Passes.td`
  - Pass definition with description and statistics
- `src/Dialect/ONNX/Transforms/CMakeLists.txt`
  - Build configuration for OMExpandAttentionMask library
- `src/Pass/Passes.hpp`
  - Function declaration for createExpandAttentionMaskPass()
- `src/Tools/onnx-mlir-opt/RegisterPasses.cpp`
  - Pass registration in onnx-mlir-opt tool

**NNPA Integration:**
- `src/Accelerators/NNPA/Compiler/NNPACompilerOptions.cpp`
  - Added `--nnpa-disable-expand-attention-mask` option
- `src/Accelerators/NNPA/Compiler/NNPACompilerOptions.hpp`
  - Option declaration
- `src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp`
  - Integrated pass into NNPA pipeline before 3-iteration optimization loop

**Testing:**
- `test/mlir/onnx/onnx_expand_attention_mask.mlir`
  - Comprehensive test suite covering:
    - Dynamic dimensions with DimAnalysis
    - Static dimensions with multiple patterns
    - Edge cases (no expansion needed, non-matching patterns)

### Key Features:
__ Handles both static and dynamic shapes
__ Integrates with existing DimAnalysis infrastructure
__ Configurable via command-line option
__ Comprehensive test coverage
__ All tests passing

### Test Results:
```
PASS: Open Neural Network Frontend :: onnx/onnx_expand_attention_mask.mlir (1 of 1)
```

### CODE REVIEW

File: src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp [hunk 1]
```diff
@@ -112,6 +112,11 @@ void addONNXToZHighPasses(mlir::PassManager &pm) {
   // Determine if Conv to Im2Col+MatMul decomposition should be enabled.
   bool enableConvToMatmul = !disableConvToMatmul;
 
+  // Expand attention masks BEFORE the loop (if not disabled).
+  // This pass uses DimAnalysis which is expensive, so run it only once.
+  if (!nnpaDisableExpandAttentionMask)
```
Comment (chentong319) [inline src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp:117]: This is before the Attention Op is lowered to onnx. Too early if there is an AttentionOp?
Should I change the Attention lowering to create  an "AttentionMask" op with a new op or a fused op? Then zdnnx library could be used?

File: src/Dialect/ONNX/Transforms/ExpandAttentionMask.cpp [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    RankedTensorType shapeType = RankedTensorType::get(
+        {static_cast<int64_t>(targetShape.size())}, builder.getI64Type());
+    Value shapeValue = onnxBuilder.concat(shapeType, shapeParts, /*axis=*/0);
+
+    // Create Expand operation.
+    // Output type is same as target type.
+    Value expandedMask = onnxBuilder.expand(targetType, maskTensor, shapeValue);
+
+    // Replace all uses in Add operations.
+    for (ONNXAddOp addOp : uses) {
```
Comment (chentong319) [inline src/Dialect/ONNX/Transforms/ExpandAttentionMask.cpp:192]: Can A.replaceAllUsesWith(B) be used?

**Review by chentong319 — APPROVED** [review]
LGTM!

File: src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp (hunk 1 above)
Comment (tungld) [reply src/Accelerators/NNPA/Compiler/NNPACompilerUtils.cpp:117]: Yes, or annotate AttentionOp with an attribute to say the mask is not broadcasting.

File: src/Dialect/ONNX/Transforms/ExpandAttentionMask.cpp (hunk 2 above)
Comment (tungld) [reply src/Dialect/ONNX/Transforms/ExpandAttentionMask.cpp:192]: Good catch! Will be back to this after holiday.

Comment (tungld) [reply src/Dialect/ONNX/Transforms/ExpandAttentionMask.cpp:192]: @chentong319 I checked again, the comment was confusing. This actually does in-place update of the 2nd operand of onnx.Add with the expanded mask (new shape) while `replaceAllUsesWith` requires shape compatible. So, it would be ok as it is.
